read-fonts/generated/generated_ift.rs

```diff
@@ -961,62 +961,29 @@ impl EntryDataMarker {
         Some(start..start + u8::RAW_BYTE_LEN)
     }
     fn codepoint_data_byte_range(&self) -> Range<usize> {
-        let start = self
-            .patch_encoding_byte_range()
-            .map(|range| range.end)
-            .unwrap_or_else(|| {
-                self.entry_id_delta_byte_range()
-                    .map(|range| range.end)
-                    .unwrap_or_else(|| {
-                        self.copy_indices_byte_range()
-                            .map(|range| range.end)
-                            .unwrap_or_else(|| {
-                                self.copy_count_byte_range()
-                                    .map(|range| range.end)
-                                    .unwrap_or_else(|| {
-                                        self.design_space_segments_byte_range()
-                                            .map(|range| range.end)
-                                            .unwrap_or_else(|| {
-                                                self.design_space_count_byte_range()
-                                                    .map(|range| range.end)
-                                                    .unwrap_or_else(|| {
-                                                        self.feature_tags_byte_range()
-                                                            .map(|range| range.end)
-                                                            .unwrap_or_else(|| {
-                                                                self.feature_count_byte_range()
-                                                                    .map(|range| range.end)
-                                                                    .unwrap_or_else(|| {
-                                                                        self.format_byte_range().end
-                                                                    })
-                                                            })
-                                                    })
-                                            })
-                                    })
-                            })
-                    })
-            });
+        let start = self . patch_encoding_byte_range () . map (| range | range . end) . unwrap_or_else (|| self . entry_id_delta_byte_range () . map (| range | range . end) . unwrap_or_else (|| self . copy_indices_byte_range () . map (| range | range . end) . unwrap_or_else (|| self . copy_count_byte_range () . map (| range | range . end) . unwrap_or_else (|| self . design_space_segments_byte_range () . map (| range | range . end) . unwrap_or_else (|| self . design_space_count_byte_range () . map (| range | range . end) . unwrap_or_else (|| self . feature_tags_byte_range () . map (| range | range . end) . unwrap_or_else (|| self . feature_count_byte_range () . map (| range | range . end) . unwrap_or_else (|| self . format_flags_byte_range () . end)))))))) ;
```

Comment:
so you've drifted too far to the right and rustfmt has left you. Can we figure out a different way to write this?

Reply:
Since this is generated code, does it matter if the formatting isn't right?

Reply:
it IS nice to be able to read it at times

Reply:
This looks to be a result of how conditional fields are handled, the codegen has created a giant if else chain to figure out which field preceeds this one. So if we want to fix it would probably require reworking how conditional fields are handled in codegen.